packages/api/lib/injectConceptId.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const cmrjs = require('@cumulus/cmrjs');
+
+/**
+ * Given a collection record with a name and version, search CMR for a single
+ * matching collection and return the concept-id if it exists.
+ * @param {Object} collectionRecord - cumulus collection record.
+ * @returns {string} CMR's concept-id for the record, or null.
+ */
+const getConceptId = async (collectionRecord) => {
+  const searchParams = {
+    short_name: collectionRecord.name || '',
+    version: collectionRecord.version || '',
+    provider_short_name: process.env.cmr_provider
+  };
+
+  const cmrResult = await cmrjs.searchConcept('collections', searchParams, []);
+
+  if (cmrResult.length !== 1) return null;
+  return cmrResult[0].id;
+};
+
+/**
+ * Add concept-id to collection record
+ * @param {Array} collections - array of collection results
+ * @param {function} conceptIdFunction - function to use to return a
+ *                   conceptId. Defaults to getConceptId.
+ * @returns {Array} - input array with each element updated with its found concept-id or null.
+ */
+const injectConceptId = async (collections, conceptIdFunction = getConceptId) => {
```

Comment:
Is passing in the concept id function just for the tests? You could also accomplish that by stubbing.

Reply:
Yes, it's just for tests and to be more functional. I would like to see how you would accomplish this by simple stubbing.  I actually tried for quite a while.   The only way I could find to replace the function was to change  L32 to `const conceptIds = collections.results.map(modules.exports.conceptIdFunction);`  or to use a third party library like proxyquire.  (as a note I think testdouble handles this quite nicely) . 

I'm all ears if I just don't know how to do this correctly.  Please show me.